tests/recorder_tests.cc

```diff
@@ -56,6 +56,15 @@ TEST(RecorderTest, Manager_GetSetDumpFreq) {
   EXPECT_EQ(m.dump_count(), cyclus::kDefaultDumpCount);
 }
 
+TEST(RecorderTest, InjectSimId) {
```

Comment:
This test needs to check whether a recorded datum actually has a sim id injected or not.  Such a test would have caught the missing `set_dump_count` call in the `inject_sim_id(yourval)` function.